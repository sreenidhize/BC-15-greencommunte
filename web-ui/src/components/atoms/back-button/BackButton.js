import React from "react";
import PropTypes from "prop-types";
import { makeStyles } from "@material-ui/core";
import ArrowBack from "@material-ui/icons/ArrowBack";

const useStyles = makeStyles((theme) => ({
  backButton: {
    display: "flex",
    alignItems: "center",
    cursor: "pointer",
  },
  backIcon: {
    color: theme.palette.secondary.main,
    paddingRight: theme.spacing(1),
    fontSize: theme.spacing(4),
  },
  buttonLabel: {
    color: theme.palette.secondary.main,
    fontFamily: theme.typography.fontFamily,
    fontWeight: "500",
    fontSize: theme.spacing(4),
  },
}));

const BackButton = ({ label, onClick }) => {
  const classes = useStyles();
  return (
    <div
      id="backButton"
      className={classes.backButton}
      onClick={onClick}
      data-testid="back-button"
    >
      <ArrowBack className={classes.backIcon} />
      <span className={classes.buttonLabel}>{label}</span>
    </div>
  );
};

BackButton.propTypes = {
  label: PropTypes.string,
  onClick: PropTypes.func.isRequired,
};

export default BackButton;
