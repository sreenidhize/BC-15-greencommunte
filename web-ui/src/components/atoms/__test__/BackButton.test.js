import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import BackButton from "../back-button/BackButton";

describe("Back button", () => {
    it("Matches the snapshot", () => {
        const { container } = render(
            <BackButton onClick={()=>{}}/>
        );
        expect(container).toMatchSnapshot();
    });

    test("test back button render", () => {
        const { getByTestId } = render(<BackButton onClick={()=>{}}/>);
        expect(getByTestId("back-button")).toBeInTheDocument();
    });
});