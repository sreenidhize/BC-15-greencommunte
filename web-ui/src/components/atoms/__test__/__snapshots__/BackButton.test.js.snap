// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Back button Matches the snapshot 1`] = `
<div>
  <div
    class="makeStyles-backButton-1"
    data-testid="back-button"
    id="backButton"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root makeStyles-backIcon-2"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
      />
    </svg>
    <span
      class="makeStyles-buttonLabel-3"
    />
  </div>
</div>
`;
